libs/core/langchain_core/messages/modifier.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, List, Literal
+
+from langchain_core.messages.base import BaseMessage, BaseMessageChunk
+
+
+class ModifierMessage(BaseMessage):
+    """Message responsible for modifying other messages (deleting / updating.)"""
+
+    def __init__(self, id: str, **kwargs: Any) -> None:
+        return super().__init__("modifier", id=id)
+
+
+class RemoveMessage(ModifierMessage):
+    """Message responsible for deleting other messages."""
+
+    type: Literal["remove"] = "remove"
+
+    @classmethod
+    def get_lc_namespace(cls) -> List[str]:
+        """Get the namespace of the langchain object."""
+        return ["langchain", "schema", "messages"]
+
+
+RemoveMessage.update_forward_refs()
+
+
+class RemoveMessageChunk(RemoveMessage, BaseMessageChunk):
```

Comment:
what purpose does this one serve? If we do want a chunk variant it should do something smart on `+` ?

Reply:
mainly to please the `BaseChatModel` typing, as it requires `BaseMessageChunk` in streaming, otherwise doesn't really make sense i agree. i think it's fine to just use `RemoveMessage` and ignore typecheck for that line